sim/priest/shadow_word_pain.go

```diff
@@ -90,17 +90,17 @@ func (priest *Priest) getShadowWordPainConfig(rank int) core.SpellConfig {
 		},
 
 		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
-			result := spell.CalcOutcome(sim, target, spell.OutcomeMagicHit)
-			if result.Landed() {
-				spell.SpellMetrics[target.UnitIndex].Hits--
-				curTarget := target
-				for hitIndex := int32(0); hitIndex < numHits; hitIndex++ {
+			curTarget := target
+			for hitIndex := int32(0); hitIndex < numHits; hitIndex++ {
+				result := spell.CalcAndDealOutcome(sim, curTarget, spell.OutcomeMagicHit)
+				if result.Landed() {
+					spell.SpellMetrics[target.UnitIndex].Hits--
 					priest.AddShadowWeavingStack(sim, curTarget)
 					spell.Dot(curTarget).Apply(sim)
-					curTarget = sim.Environment.NextTargetUnit(curTarget)
+					spell.DealOutcome(sim, result)
 				}
+				curTarget = sim.Environment.NextTargetUnit(curTarget)
 			}
-			spell.DealOutcome(sim, result)
```

Comment:
Did someone confirm this is how the chaning of the dot happens? I had asked before if it rolls separate hits and was told only the main cast rolls hit then every other chained dot is just applied with it but it could have been wrong. Just making sure this is an intended change

Reply:
I asked yesterday and was told by Tako in the priest discord that each one rolls. I'm also pretty sure I've seen that happen in game

Reply:
The only reason I changed this is that I was implementing Burn which sounds like it works the same way